### PR TITLE
Add generic image

### PR DESCRIPTION
**Pull request by ufoscout**
This is an attempt to solve  #56

It introduces a GenericImage implementation and a generic way of handling WaitFor strategies.

### CODE REVIEW

File: images/generic/src/image.rs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+pub enum WaitFor {
+    Nothing,
+    LogMessage(String),
+}
+
+impl WaitFor {
+    fn wait<D: Docker, I: Image>(&self, container: &Container<D, I>) -> Result<(), WaitError> {
+        match self {
+            WaitFor::Nothing => Ok(()),
+            WaitFor::LogMessage(message) => container.logs().stderr.wait_for_message(message),
```
Comment (thomaseizinger) [inline images/generic/src/image.rs]: Not sure if hardcoding `stderr` here is a good idea.

The other thing is, a user would currently have to call `to_owned` on a static string but most times, what they will pass is a `&'static str`.

Maybe we can kill two birds with one stone here and create factory functions on `WaitFor` to initialize it from a `&str` and at the same time configure a flag, which stream the message is expected on.

```rust
impl WaitFor {

    fn message_on_stdout(message: &str) -> Self {
        Self::LogMessage {
            message: message.to_owned(),
            stream: Stream::StdOut
        }
    }


    fn message_on_stderr(message: &str) -> Self {
        Self::LogMessage {
            message: message.to_owned(),
            stream: Stream::StdErr
        }
    }
}
```

Comment (ufoscout) [reply images/generic/src/image.rs]: @thomaseizinger  
Fixed

**Comment by ufoscout** [comment]
@thomaseizinger 
Comments fixed. Let me know if it's fine.

File: images/generic/README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+is not provided by `testcontainers`. 
+
+E.g.:
+```rust
+// Creates a generic image with descriptor "postgres:9.6-alpine"
+let generic_postgres = images::generic::GenericImage::new("postgres:9.6-alpine")
+    // Instructs testcontainers to wait until the LogMessage
+    // "database system is ready to accept connections" is printed in the
+    // container system output
+    .with_wait_for(images::generic::WaitFor::LogMessage(
```
Comment (thomaseizinger) [inline images/generic/README.md]: The documentation is now outdated :sweat_smile: 

I'd say remove the whole block in favor of a link to the integration tests or rust-doc.

**Comment by ufoscout** [comment]
@thomaseizinger 
Outdated code replaced with a link to the integration tests.

**Pull request merged**